src/internal/util/reportUnhandledError.ts

```diff
@@ -11,7 +12,7 @@ import { config } from '../config';
  * @param err the error to report
  */
 export function reportUnhandledError(err: any) {
-  setTimeout(() => {
+  immediateProvider.setImmediate(() => {
```

Comment:
issue: Errors thrown in this user-provided function (`onUnhandledError`) will be unhandled promise rejections, because `setImmediate` uses a resolved promise to schedule by default. We should probably do this on `setTimeout`, as it will result in errors being sent to `window.onerror` or `process.on('error', fn)` as I think more people will expect.

Reply:
Added `timeoutProvider`, etc.